src/codec.rs

```diff
@@ -113,6 +131,108 @@ pub trait Decode: Sized {
 /// Trait that allows zero-copy read/write of value-references to/from slices in LE format.
 pub trait Codec: Decode + Encode {}
 
+/// Compact-encoded variant of T. This is more space-efficient but less compute-efficient.
+pub struct Compact<T>(T);
```

Comment:
I think we probably want to make the field `pub`. Otherwise the users of the crate won't be able to crate instances of it nor read the value.